File: frontend/src/types/model-config.ts
// 模型配置类型定义

export interface ModelConfig {
  id: number;
  name: string;
  model_id: string;
  description: string;
  provider: string;
  icon: 'sparkles' | 'zap' | 'brain' | 'cpu' | 'bot';
  icon_color: string;
  max_tokens: number;
  api_base_url?: string;
  api_key?: string;
  api_key_env?: string;
  is_active: boolean;
  sort_order: number;
  created_at: string;
  updated_at: string;
}

// 公开接口返回的模型配置（不包含敏感信息）
export interface ModelConfigPublic {
  id: number;
  name: string;
  model_id: string;
  description: string;
  provider: string;
  icon: 'sparkles' | 'zap' | 'brain' | 'cpu' | 'bot';
  icon_color: string;
  max_tokens: number;
}

// 创建/更新模型配置的请求体
export interface ModelConfigInput {
  name: string;
  model_id: string;
  description?: string;
  provider: string;
  icon?: string;
  icon_color?: string;
  max_tokens?: number;
  api_base_url?: string;
  api_key?: string;
  api_key_env?: string;
  is_active?: boolean;
  sort_order?: number;
}

// Provider 选项
export const PROVIDER_OPTIONS = [
  { value: 'openai', label: 'OpenAI' },
  { value: 'anthropic', label: 'Anthropic' },
  { value: 'deepseek', label: 'DeepSeek' },
  { value: 'alibaba', label: 'Alibaba' },
  { value: 'siliconflow', label: 'SiliconFlow' },
  { value: 'other', label: 'Other' }
];

// Icon 选项
export const ICON_OPTIONS = [
  { value: 'sparkles', label: 'Sparkles ✨' },
  { value: 'zap', label: 'Zap ⚡' },
  { value: 'brain', label: 'Brain 🧠' },
  { value: 'cpu', label: 'CPU 💻' },
  { value: 'bot', label: 'Bot 🤖' }
];

// 图标颜色选项
export const ICON_COLOR_OPTIONS = [
  { value: 'text-blue-500', label: '蓝色' },
  { value: 'text-emerald-500', label: '绿色' },
  { value: 'text-violet-500', label: '紫色' },
  { value: 'text-orange-500', label: '橙色' },
  { value: 'text-amber-500', label: '琥珀色' },
  { value: 'text-cyan-500', label: '青色' },
  { value: 'text-purple-500', label: '深紫色' },
  { value: 'text-rose-500', label: '玫红色' }
];

File: backend/app/models/model_config.py
"""
模型配置数据库模型
"""

from sqlalchemy import Column, Integer, String, Boolean, Text, DateTime
from sqlalchemy.sql import func
from app.core.database import Base


class ModelConfig(Base):
    """
    模型配置模型
    """

    __tablename__ = "model_configs"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(200), nullable=False, index=True, comment="模型名称")
    model_id = Column(String(200), nullable=False, index=True, comment="模型标识")
    description = Column(Text, nullable=True, comment="模型描述")
    provider = Column(
        String(50),
        nullable=False,
        default="siliconflow",
        comment="供应商：openai/anthropic/deepseek/alibaba/siliconflow/other",
    )
    icon = Column(
        String(50),
        nullable=True,
        default="sparkles",
        comment="图标：sparkles/zap/brain/cpu/bot",
    )
    icon_color = Column(
        String(50), nullable=True, default="text-blue-500", comment="图标颜色"
    )
    max_tokens = Column(Integer, nullable=False, default=4096, comment="最大token数")
    api_base_url = Column(String(500), nullable=True, comment="API基础URL")
    api_key = Column(Text, nullable=True, comment="API密钥（加密存储）")
    api_key_env = Column(String(200), nullable=True, comment="API密钥环境变量名")
    is_active = Column(
        Boolean, nullable=False, default=True, index=True, comment="是否启用"
    )
    sort_order = Column(Integer, nullable=False, default=0, index=True, comment="排序")
    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
        comment="创建时间",
    )
    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
        comment="更新时间",
    )

    def __repr__(self):
        return f"<ModelConfig(id={self.id}, name='{self.name}', model_id='{self.model_id}')>"

